portfolio/src/main/webapp/leave-comment/script.js

```diff
@@ -1,8 +1,25 @@
+function onSignIn(googleUser) {
+  $('#userTokenId').val(googleUser.getAuthResponse().id_token);
+  $('#show-on-sign-in')
+      .css('visibility', 'visible');
+  $('#greeting').text('Welcome, ' + googleUser.getBasicProfile().getName() + '!');
+}
+
+function signOut() {
+  var auth2 = gapi.auth2.getAuthInstance();
+  auth2.signOut().then(function () {
+    console.log('User signed out.');
+  });
+  $('#show-on-sign-in')
```

Comment:
Note that signing out can also happen without user's explicit request on your page (e.g. the user gets signs out of Google on another page). If you [handle the state changes in a listener](https://developers.google.com/identity/sign-in/web/listeners), your webapp's state can be kept consistent with the signin state at all times. Also, you get to handle the state changes in a centralized place so it's easier to avoid...

Reply:
Done! :)